x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/OptimizerRulesTests.java

```diff
@@ -143,4 +179,306 @@ public void testOrWithNonCombinableExpressions() {
         assertEquals(fa, in.value());
         assertThat(in.list(), contains(ONE, THREE));
     }
+
+    // a == 1 AND a == 2 -> FALSE
+    public void testDualEqualsConjunction() {
+        FieldAttribute fa = getFieldAttribute();
+        Equals eq1 = equalsOf(fa, ONE);
+        Equals eq2 = equalsOf(fa, TWO);
+
+        OptimizerRules.PropagateEquals rule = new OptimizerRules.PropagateEquals();
+        Expression exp = rule.rule(new And(EMPTY, eq1, eq2));
+        assertEquals(FALSE, exp);
+    }
+
+    // a <=> 1 AND a <=> 2 -> FALSE
+    public void testDualNullEqualsConjunction() {
+        FieldAttribute fa = getFieldAttribute();
+        NullEquals eq1 = nullEqualsOf(fa, ONE);
+        NullEquals eq2 = nullEqualsOf(fa, TWO);
+
+        OptimizerRules.PropagateEquals rule = new OptimizerRules.PropagateEquals();
+        Expression exp = rule.rule(new And(EMPTY, eq1, eq2));
+        assertEquals(FALSE, exp);
+    }
+
+    // 1 < a < 10 AND a == 10 -> FALSE
+    public void testEliminateRangeByEqualsOutsideInterval() {
+        FieldAttribute fa = getFieldAttribute();
+        Equals eq1 = equalsOf(fa, new Literal(EMPTY, 10, DataTypes.INTEGER));
+        Range r = rangeOf(fa, ONE, false, new Literal(EMPTY, 10, DataTypes.INTEGER), false);
```

Comment:
Even though we don't have explicit ranges in ES|QL, these tests seemed worth keeping, in case optimization (current or future) rewrite other comparisons into Ranges.